File: 75-Sort-Colors.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        int l=0,r=nums.size()-1;
        for (int i=0;i<=r;i++){
            if (nums[i]==0) {
                nums[i]=nums[l];
                nums[l++]=0;
            }
            else if (nums[i]==2) {
                nums[i--]=nums[r];
                nums[r--]=2;
            }
        }
    }
};

File: 75-Sort-Colors.java
class Solution {
    public void sortColors(int[] nums) {
        
        int n = nums.length;
        int l=-1;

        // Sorting twos only
        for (int i=0; i<n; i++){
            if (nums[i] < 2){
                l++;
                swap(nums, l, i);
            }
        }

        // Sorting zeros and ones only
        l = -1;
        for (int i=0; i<n; i++){
            if (nums[i] < 1){
                l++;
                swap(nums, l, i);
            }
            else if (nums[i] == 2){
                break;
            }
        }

        // Another solution: Frequency array then overrite the original one
    }

    void swap(int[] arr, int i, int j){
        int temp = arr[i];
        arr[i] = arr[j];
        arr[j] = temp;
    }
}